openedx/openedx-learning
# Taxonomy import templates [FC-0036]

This PR adds CSV and JSON templates which demonstrate the format for Taxonomy Administrators to use when importing the tags to a taxonomy.

The templates use an ignored "comments" fields to explain how to use the template inline with the example. To support this, I altered the taxonomy import code to ignore any unexpected fields. Ignoring unexpected fields will also make it easier for users to import files provided by external taxonomy suppliers, as these are likely to contain more fields than we need.

Also adds tests to ensure that the templates import without error.

Closes https://github.com/openedx/modular-learning/issues/137

Private-ref: [FAL-3520](https://tasks.opencraft.com/browse/FAL-3520)

**Testing instructions**

1. Using this branch, run the dev server: `python manage.py runserver` 
2. Download the template files:
    http://127.0.0.1:8000/tagging/rest_api/v1/import/template.csv
    http://127.0.0.1:8000/tagging/rest_api/v1/import/template.json
3. Check that the files download as expected.

**TO DO BEFORE MERGE**

- [x] Bump version to 0.2.2
